src/registers/model_specific.rs

```diff
@@ -6,6 +6,7 @@ use bitflags::bitflags;
 use crate::registers::segmentation::{FS, GS};
 
 /// A model specific register.
+#[cfg_attr(not(feature = "instructions"), allow(dead_code))] // FIXME
```

Comment:
What is this FIXME about? The `allow(dead_code)`?

Reply:
Yeah, this FIXME and the other one are about the field in this struct never being read without `feature = "instructions"` (modulo the `#[derive(Debug)]`).

That lead me to believe that maybe these types don't make sense without `feature = "instructions"` and should be `#[cfg(feature = "instructions")]` themselves, but that would be a breaking and non-trivial change.

Would you like me to elaborate in this FIXME about its reasons, or should I open an issue about this, or should I just remove the FIXME? :D

Reply:
Maybe we shouldn't set `RUSTFLAGS: -Dwarnings` globally, but only for the [clippy](https://github.com/rust-osdev/x86_64/blob/2eb838d101cdd179d6395f79f87916a48e8c1935/.github/workflows/build.yml#L226) invocation. That way we wouldn't fail for the build that disable some of the default features (as is the case here).

Reply:
> Maybe we shouldn't set `RUSTFLAGS: -Dwarnings` globally, but only for the [clippy](https://github.com/rust-osdev/x86_64/blob/2eb838d101cdd179d6395f79f87916a48e8c1935/.github/workflows/build.yml#L226) invocation.

That's definitely one possibility. There is benefit in checking for warnings in different configurations, possibly highlighting that more code should be `#[cfg]`'d out, but if you don't deem it important enough, I can surely do as you suggest. :)

Reply:
I like having `-Dwarnings` for all configurations, as it alters us to the fact that our code might have incorrect `#[cfg]`s. For example, the need for `alloc(dead_code)` here tells me that either:
  - We should expose an `Msr::address()` function, so that users can do:
    ```rust
    const EFER_ADDRESS: u32 = Efer::MSR.address()
    ```
  - We should not expose the `Msr` type if `instructions` is disabled.